internal/api/v1/organizations.go

```diff
@@ -93,3 +96,73 @@ func (oc OrganizationsController) Create(w http.ResponseWriter, r *http.Request)
 	w.WriteHeader(http.StatusCreated)
 	w.Write([]byte{})
 }
+
+func (oc OrganizationsController) Delete(w http.ResponseWriter, r *http.Request) {
+	params := httprouter.ParamsFromContext(r.Context())
+	org := params.ByName("org")
+
+	gitea, err := gitea.New()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	cluster, err := kubernetes.GetCluster()
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	exists, err := organizations.Exists(cluster, org)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+	if !exists {
+		http.Error(w, fmt.Sprintf("Organization '%s' does not exist", org),
+			http.StatusNotFound)
+		return
+	}
+
+	apps, err := application.List(cluster, org)
+	if handleError(w, err, http.StatusInternalServerError) {
+		return
+	}
+
+	servicesToBeDeleted := []string{}
+
+	for _, app := range apps {
+		err = application.Delete(cluster, gitea, org, app)
+		if handleError(w, err, http.StatusInternalServerError) {
+			return
+		}
+
+		servicesToBeDeleted = append(servicesToBeDeleted, app.BoundServices...)
+	}
+
```

Comment:
Would it not be simpler (and more correct) to ask for all services in the org, bound or not ?

With the above it will only try to delete services in the org bound to an app.
Services in the org not bound to an app are not deleted.

Nit: If a service is bound to multiple apps the code will record it multiple times, and try to delete multiple times.
Simply asking for and deleting all services in the org trivially avoids that.

Reply:
Done